tests/mapper/test_node_mappers.py

```diff
@@ -559,3 +559,88 @@ def test_hinnodemapper_no_neighbors():
 
     # Second edge type (e2): Node 0 has 2, node 1 has none, and node 6 sampling has terminated
     assert np.all(batch_feats[3][:, 0, 0] == np.array([12, 0, 0]))
+
+
+class Test_FullBatchNodeGenerator:
+    """
+    Tests of FullBatchNodeGenerator class
+    """
+
+    n_feat = 4
+    target_dim = 5
+
+    G = example_graph_3(feature_size=n_feat, n_nodes=6, n_isolates=1, n_edges=20)
+    N = len(G.nodes())
+
+    def test_generator_constructor(self):
+        generator = FullBatchNodeGenerator(self.G)
+
+        assert generator.Aadj.shape == (self.N, self.N)
+        assert generator.features.shape == (self.N, self.n_feat)
+
+    def test_generator_constructor_wrong_G_type(self):
+        with pytest.raises(TypeError):
+            generator = FullBatchNodeGenerator(nx.Graph(self.G))
+
+    def test_generator_constructor_hin(self):
+        feature_sizes = {"t1": 1, "t2": 1}
+        Ghin, nodes_type_1, nodes_type_2 = example_hin_3(feature_sizes)
+        with pytest.raises(TypeError):
+            generator = FullBatchNodeGenerator(Ghin)
+
+    def test_generator_flow_notargets(self):
+        generator = FullBatchNodeGenerator(self.G)
+        node_ids = self.G.nodes()
+        gen = generator.flow(node_ids)
+
+        assert len(gen) == 1
+        assert gen.A.shape == (self.N, self.N)
+        assert gen.features.shape == (self.N, self.n_feat)
+        assert sum(gen.sample_weight) == len(node_ids)
+        assert gen.targets is None
+
+        [X, A], y, sample_weights = gen.__getitem__(0)
+        assert np.array_equal(X, gen.features)  # X should be equal to gen.features
+        assert (A != gen.A).nnz == 0  # A should be equal to gen.A
```

Comment:
`len(np.nonzero(A != gen.A)[0])` would work for both dense and sparse matrices :-)

Reply:
fixed